### PR TITLE
add new gzip section in compression chapter

### PR DESCRIPTION
**Pull request by sepiropht**
--------------------------
### Things to check before submitting a PR

- [x] the tests are passing locally with `cargo test`
- [x] cookbook renders correctly in `mdbook serve -o`
- [x] commits are squashed into one and rebased to latest master
- [ ] PR contains correct "fixes #ISSUE_ID" clause to autoclose the issue on PR merge
    -  if issue does not exist consider creating it or remove the clause
- [ ] spell check runs without errors `./ci/spellchecker.sh`
- [ ] link check runs without errors `link-checker ./book`
- [ ] non rendered items are in sorted order (links, reference, identifiers, Cargo.toml)
- [ ] links to docs.rs have wildcard version `https://docs.rs/tar/*/tar/struct.Entry.html`
- [ ] example has standard [error handling](https://rust-lang-nursery.github.io/rust-cookbook/about.html#a-note-about-error-handling)
- [ ] code identifiers in description are in hyperlinked backticks
```markdown
[`Entry::unpack`]: https://docs.rs/tar/*/tar/struct.Entry.html#method.unpack
```

### Things to do after submitting PR
- [ ] check if CI is happy with your PR

Thank you for reading, you may now delete this text! Thank you! :smile:

### CODE REVIEW

File: src/compression/gzip/gzip-dowload.md [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+#         ReqError(reqwest::Error);
+#         Io(std::io::Error);
+#     }
+# }
+use std::io::Read;
+use flate2::read::GzDecoder;
+
+fn main() -> Result<()> {
+    let response = reqwest::get("https://httpbin.org/gzip")?;
+    let mut d = GzDecoder::new(response);
```
Comment (budziq) [inline src/compression/gzip/gzip-dowload.md]: please note that reqwest already has [gzip related capabilities](https://docs.rs/reqwest/0.8.6/reqwest/struct.ClientBuilder.html#method.gzip) if anything, we should show them as more idiomatic example.

File: src/compression/gzip/gzip-dowload.md [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+    let mut s = String::new();
+    d.read_to_string(&mut s).unwrap();
+    println!("{}", s);
+    Ok(())
+}
+#
+```
+
+[`reqwest`]: https://docs.rs/reqwest
```
Comment (budziq) [inline src/compression/gzip/gzip-dowload.md]: the link to crate already exists, please note that we are using backticks `` to show links to functions and other symbols while reqwest is a crate name.

File: src/compression/gzip/gzip-dowload.md [hunk 3]
```diff
@@ -0,0 +1,35 @@
+## Decompress gzip response for a http request
+
+[![flate2-badge]][flate2] [![reqwest-badge]][reqwest] [![cat-decompression-badge]][cat-decompression]
+
+Donwload a content with [`reqwest`]. </br>
+[`GzEncoder`] is responsible for decompress the response.
```
Comment (budziq) [inline src/compression/gzip/gzip-dowload.md]: - 'decompress' -> 'decompression'.
- Also the description is a little off. Could you look into how the examples from tar section are described?

**Comment by sepiropht** [comment]
I have to admit no

**Comment by AndyGauge** [comment]
OK, no problem.  Maybe next week I'll take a look at getting this to the finish line.  Thank you so much for your contribution.

File: src/compression/gzip/gzip-dowload.md (hunk 1 above)
Comment (AndyGauge) [reply src/compression/gzip/gzip-dowload.md]: > 
> 
> please note that reqwest already has [gzip related capabilities](https://docs.rs/reqwest/0.8.6/reqwest/struct.ClientBuilder.html#method.gzip) if anything, we should show them as more idiomatic example.

Closing this PR and issue, there's nothing to show.

**Pull request closed without merging**